### ISSUE TITLE
BUG - sev-3 - All - Letters: Misleading toggle label on BS&SV letter when monthly award is 0 dollars

### ISSUE THREAD

**Issue by TKDickson** [department-of-veterans-affairs/va-mobile-app#5064]
## What happened?

There's a toggle in the Benefit Summary & Service Verification letter that controls whether or not to display two pieces of information, but we only list out one of them in the toggle description.

Our description:
![image.png](https://images.zenhubusercontent.com/61a5397dac83844bf079e942/d9748707-2878-4127-a9fd-81d33e919db8)

What it controls in the letter:
![image.png](https://images.zenhubusercontent.com/61a5397dac83844bf079e942/636a77e3-2c2f-4bb6-877c-3bfc995d5dae)

Website description (which lists both):
![image.png](https://images.zenhubusercontent.com/61a5397dac83844bf079e942/1e19c6f0-2376-416a-aabd-774475e4ea72)

## Specs:

- Device:
- OS:
- User Account: have to use a staging user to actually get the 'working' letter functionality - +54 has this letter, among others
- Accessibility State:

## Steps to Reproduce

Benefits > Letters > View Letters > Benefit Summary & Service Verification > toggle everything except "effective date" off > generate letter

And see that there's more in there than just effective date.

## Desired behavior

"... If there are no product/technical reasons for this omission, I'd recommend calling out both pieces of information in the toggle. So it'd be something like this:

Your current monthly award is $[#].
The effective date of the last change to your current award was [date].

Both of those lines of copy should be used to label that toggle. These 2 lines of copy should also start on separate lines (the date sentence should start on a separate line as opposed to directly after the $ sentence). If this needs a mockup, UX design can help with that part."

## Bug Severity - BE SURE TO ADD THE SEVERITY LABEL
 
See [Issue Tracking](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/va-mobile-app/testing/VA%20Mobile%20App%20Test%20Plan.md#issue-tracking) for details on Severity Levels

- 3 - Low

## Ticket Checklist
- [ ] Steps to reproduce are defined
- [ ] Desired behavior is added
- [ ] Labels added (front-end, back-end, feature, bug)
- [ ] Estimate of 1 added

**Comment by Sparowhawk** [comment]
@bischoffa @TKDickson 

This should already be taken care of. If there is a monthly reward amount and a date provided we used this string:

Your current monthly payment is ${{monthlyAwardAmount}}. The effective date of the last change to your current payment was {{date}}.

If there is only a monthly payment we use:

Your current monthly payment is ${{monthlyAwardAmount}}.

Lastly if there is only a date:

The effective date of the last change to your current award was {{date}}.

This ticket doesn't have anything to work as it is already implemented. The last time this section of code was touched was 2 years ago. So if you are still seeing this issue in staging it likely is a data issue coming to us.

**Comment by TKDickson** [comment]
Oh. Wait. Tom already called this out in his earlier comment:

> I can reproduce this issue with the +54 user called out in the description. [...]
> It appears that when we get "monthlyAwardAmount": 0.0, , we show a different message

@Sparowhawk I'll update the ticket title accordingly. @bischoffa idk how realistic this is/if we want to fix to be in sync with the website or not.

**Comment by bischoffa** [comment]
@TKDickson @Sparowhawk  based on the comments above is this "bug" more of a call out that there is a discrepancy between web vs mobile?   
  
Is mobile working as we expected? (ignore what web does for this question).

**Comment by TKDickson** [comment]
@htcollier or @mistymg, can one of you weigh in on Adam's question? Ignoring the web/mobile discrepancy, is the toggle label working the way we'd like it to on mobile (where a $0 monthly award drops that portion of the label from the toggle). 

_I have an opinion, which I am holding onto by sheer willpower.... must not share opinion until after design has decided......_

**Comment by mistymg** [comment]
Sorry this is sooooo old it's taking me awhile to wrap my head around. 

I think if the monthly award is $0, we should still show that amount...even though it's zero. A follow-up question I have is what does web show when there is no date? Does web drop that part of the copy? Will there EVER be no date?

**Comment by htcollier** [comment]
@bischoffa Based on the description above, mobile is not working as I'd expect in this case because the toggle label isn't accurately indicating to the user what the impact of their actions will be when generating the letter. 
  
**tl;dr** So long as flipping this toggle results in displaying 2 lines of text in the generated letter, we should not drop any portion of the toggle label based on the field data, because the label would not accurately describe the impact of the toggle action. The copy @mistymg provided looks good to me.   _*If the letter we're displaying on mobile is not being generated correctly, that would change the approach here._

**More on that:** 
- If the toggle **_always_** turns on two different lines of text in the letter (including the one that says $0 for the monthly award), the label next to the toggle should accurately reflect the function of the toggle by **_always_** displaying both of those pieces of information (what we're doing now incorrectly implies that we'll only show the sentence featured, and the monthly award line appearing in the letter may come as a surprise).
- If there is/is intended to be variability in what gets generated in the letter based on the monthly amount, then I could see adjusting the content structure of the toggle to set _those_ expectations, but that doesn't seem to be the case based on what's described above.

**Comment by Sparowhawk** [comment]
@mistymg there is a scenario we have coded where there isn't a date provided, the same without the $ amount for the monthly award. I agree that we probably should include both lines no matter if we have data or not. The question becomes what should be the copy for those fields be?

Your current monthly payment is "dollar amount or 'not provided'". The effective date of the last change to your current payment was "date or 'not provided'"?

**Comment by mistymg** [comment]
Ok, agree whether or not we have the data, the copy should be shown for both lines on the toggle.

If we need to know what copy should appear if there's no data, we should follow whatever web is doing. So, what does their checkbox say when there's no data?

**Comment by TKDickson** [comment]
For benefits-and-adjacent, I ask @StacyB2023 or @kellylein if they know the right people to ping

**Comment by StacyB2023** [comment]
@Sparowhawk @TKDickson Talked this over with @kellylein. We are thinking the best contact would be Jacob Worrell on the benefits side or he will direct you to the best person on his team.

**Comment by Sparowhawk** [comment]
@StacyB2023 @TKDickson Do either of you know how to contact him or what channel in DSVA I should ask?

**Comment by Sparowhawk** [comment]
@TKDickson @bischoffa @StacyB2023 After talking with Jacob it turns out web displays $0 if the value is null for the amount and if the date is null it displays 'invalid date'

Will make the adjustment so mobile matches web

**Comment by rbontrager** [comment]
Verified that the toggle label is correct when the monthly award is 0 dollars or there is an invalid date.  Approved by QA.
